### PR TITLE
Add some more specific PHPDoc type annotations

### PR DESCRIPTION
**Pull request by oliverklee**
This will help static analysis for code that uses this library.

### CODE REVIEW

File: src/TRegx/DataProvider/CrossDataProviders.php [hunk 1]
```diff
@@ -3,6 +3,11 @@
 
 class CrossDataProviders
 {
+    /**
+     * @param array<array-key, array<int, mixed>> ...$dataProviders
```
Comment (oliverklee) [inline src/TRegx/DataProvider/CrossDataProviders.php:7]: I'm not quite sure whether the types for parameters with the splat operator are correct, though. Could you please double-check when reviewing? Thanks!

Comment (Danon) [reply src/TRegx/DataProvider/CrossDataProviders.php:7]: What IDE are you using? Is it PhpStorm?

Comment (oliverklee) [reply src/TRegx/DataProvider/CrossDataProviders.php:7]: I'm using PhpStorm, and for https://github.com/MyIntervals/emogrifier, we're using Psalm for static type analysis.

Comment (oliverklee) [reply src/TRegx/DataProvider/CrossDataProviders.php:7]: (PHPStan also recognizes the array<int, foo> syntax.)

Comment (oliverklee) [reply src/TRegx/DataProvider/CrossDataProviders.php:7]: And I'm using the "PHP Inspections (EA Ultimate)" (i.e., the paid version of the similarly-named free plugin) PhpStorm plugin for more powerful static analysis in PhpStorm.

Comment (Danon) [reply src/TRegx/DataProvider/CrossDataProviders.php:7]: @oliverklee I think conforming to Psalm recommendations and syntax is the way to go. @iquito What so you think?

File: src/TRegx/DataProvider/CrossDataProviders.php [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+     *
+     * @return array<string, array<int, mixed>>
```
Comment (oliverklee) [inline src/TRegx/DataProvider/CrossDataProviders.php:9]: Thinking of this: I think it needs to be `int|string`, not just `string`. Change coming up in a minute.

Comment (oliverklee) [reply src/TRegx/DataProvider/CrossDataProviders.php:9]: Ah, no, for the `@return`, the keys always are strings. But I've made the `@param` annotations a bit more specific and repushed.

**Pull request merged**